### PR TITLE
[VTK] VTK ExprTk cmake patch

### PR DESCRIPTION
**Pull request by ArashPartow**
This patch is intended to unblock the following PR: https://github.com/microsoft/vcpkg/pull/29665

and is based on this VTK MR: https://gitlab.kitware.com/vtk/vtk/-/merge_requests/9981

### CODE REVIEW

**Comment by ArashPartow** [comment]
@ LilyWangLL  / @JonLiu1993 / @FrankXie05  when you have a moment can you please review and merge this PR.

**Comment by dg0yt** [comment]
Past PRs: #29775, #29665.
It is probably easier  if you don't re-start a new PR after each change. Just add new commits with the requested changes.

**Comment by ArashPartow** [comment]
@dg0yt  The PR https://github.com/microsoft/vcpkg/pull/29665 is awaiting this PR to be merged - it has to remain a seperate PR as it is for another project.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ArashPartow** [comment]
@JavierMatosD  Looks like "microsoft.vcpkg.pr (x86_windows) " needs to be rerun, I don't have perms to rerun that part, can you please trigger it.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ArashPartow** [comment]
@JonLiu1993  / @ LilyWangLL / @FrankXie05   if there's nothing else can this be merged as the following PR depends on this being merged first: https://github.com/microsoft/vcpkg/pull/29665

File: ports/vtk/FindExprTk.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
++++ b/CMake/FindExprTk.cmake
+@@ -1,14 +1,14 @@
+ # - Try to find ExprTk lib
+ #
+ # This module supports requiring a minimum version, e.g. you can do
+-#   find_package(ExprTk 2.71828)
+-# to require version 2.71828 or newer of ExprTk.
++#   find_package(ExprTk 2.7)
++# to require version 2.7 or newer of ExprTk.
+ #
```
Comment (JavierMatosD) [inline ports/vtk/FindExprTk.patch:13]: This find_package call needs to be patched out to remove the version constraint entirely.

File: ports/vtk/FindExprTk.patch [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
++      set(ExprTk_VERSION "${ExprTk_VERSION}${CMAKE_MATCH_1}")
++    endif ()
++    if (_exprtk_version_match MATCHES "\;")
++      break()
++    endif ()
++  endforeach ()
++  if (NOT ExprTk_VERSION)
++    # fallback: version in exprtk.hpp has always started with 2.7
++    set(ExprTk_VERSION "2.7")
++  endif ()
```
Comment (JavierMatosD) [inline ports/vtk/FindExprTk.patch:50]: Is there a reason we need to know the version? I believe this should be patched out to remove any version constraints.

**Review by JavierMatosD — CHANGES_REQUESTED** [review]

File: ports/vtk/FindExprTk.patch (hunk 2 above)
Comment (ArashPartow) [reply ports/vtk/FindExprTk.patch:50]: @JavierMatosD  I'm not sure about the answer to that as I am not familiar with the VTK build setup. For more details there is a discussion here: https://gitlab.kitware.com/vtk/vtk/-/merge_requests/9981   

My objective is to get this PR merged to unblock the following PR: https://github.com/microsoft/vcpkg/pull/29665   that has been around for some time now. As there are several projects I know of that use ExprTk via vcpkg and do not require/depend on VTK that are waiting on the version bump. 

@JavierMatosD  in short I'm happy for you to improve this particular cmake as a separate PR.

----------------------------------------

@JonLiu1993  If there is nothing else that needs doing can we please **merge** this PR.

File: ports/vtk/FindExprTk.patch (hunk 1 above)
Comment (ArashPartow) [reply ports/vtk/FindExprTk.patch:13]: That is a comment in the CMake.

**Pull request closed without merging**